File: sdks/elixir/lib/hooksniff/resources/endpoints.ex
defmodule HookSniff.Endpoints do
  @moduledoc "Manage webhook endpoints — create, list, update, delete, rotate secrets."

  alias HookSniff.Client

  @doc "List all endpoints"
  @spec list(HookSniff.t()) :: {:ok, list()} | {:error, term()}
  def list(client), do: Client.request(:get, "/v1/endpoints", nil, client)

  @doc "Create a new endpoint"
  @spec create(HookSniff.t(), map()) :: {:ok, map()} | {:error, term()}
  def create(client, params), do: Client.request(:post, "/v1/endpoints", params, client)

  @doc "Get endpoint by ID"
  @spec get(HookSniff.t(), String.t()) :: {:ok, map()} | {:error, term()}
  def get(client, id), do: Client.request(:get, "/v1/endpoints/#{id}", nil, client)

  @doc "Update an endpoint"
  @spec update(HookSniff.t(), String.t(), map()) :: {:ok, map()} | {:error, term()}
  def update(client, id, params), do: Client.request(:put, "/v1/endpoints/#{id}", params, client)

  @doc "Delete an endpoint"
  @spec delete(HookSniff.t(), String.t()) :: {:ok, map()} | {:error, term()}
  def delete(client, id), do: Client.request(:delete, "/v1/endpoints/#{id}", nil, client)

  @doc "Rotate the signing secret for an endpoint"
  @spec rotate_secret(HookSniff.t(), String.t()) :: {:ok, map()} | {:error, term()}
  def rotate_secret(client, id), do: Client.request(:post, "/v1/endpoints/#{id}/rotate-secret", %{}, client)
end

File: sdks/csharp/src/HookSniff/Resources/Endpoints.cs
using HookSniff.Model;

namespace HookSniff.Resources;

/// <summary>
/// Endpoints resource — manage webhook endpoints.
/// </summary>
public class Endpoints
{
    private readonly RequestContext _ctx;

    internal Endpoints(RequestContext ctx) => _ctx = ctx;

    /// <summary>List all endpoints.</summary>
    public async Task<List<Endpoint>> ListAsync()
    {
        var req = new Request("GET", "/v1/endpoints");
        var resp = await req.SendAsync<EndpointListResponse>(_ctx);
        return resp?.Data ?? new List<Endpoint>();
    }

    /// <summary>Get a single endpoint by ID.</summary>
    public async Task<Endpoint> GetAsync(string endpointId)
    {
        var req = new Request("GET", "/v1/endpoints/{endpoint_id}");
        req.SetPathParam("endpoint_id", endpointId);
        return (await req.SendAsync<Endpoint>(_ctx))!;
    }

    /// <summary>Create a new endpoint.</summary>
    public async Task<Endpoint> CreateAsync(CreateEndpointRequest body)
    {
        var req = new Request("POST", "/v1/endpoints");
        req.SetBody(body);
        return (await req.SendAsync<Endpoint>(_ctx))!;
    }

    /// <summary>Update an endpoint.</summary>
    public async Task<Endpoint> UpdateAsync(string endpointId, UpdateEndpointRequest body)
    {
        var req = new Request("PUT", "/v1/endpoints/{endpoint_id}");
        req.SetPathParam("endpoint_id", endpointId);
        req.SetBody(body);
        return (await req.SendAsync<Endpoint>(_ctx))!;
    }

    /// <summary>Delete an endpoint.</summary>
    public async Task DeleteAsync(string endpointId)
    {
        var req = new Request("DELETE", "/v1/endpoints/{endpoint_id}");
        req.SetPathParam("endpoint_id", endpointId);
        await req.SendVoidAsync(_ctx);
    }

    /// <summary>Rotate the signing secret for an endpoint.</summary>
    public async Task<EndpointsIdRotateSecretPost200Response> RotateSecretAsync(string endpointId)
    {
        var req = new Request("POST", "/v1/endpoints/{endpoint_id}/rotate-secret");
        req.SetPathParam("endpoint_id", endpointId);
        return (await req.SendAsync<EndpointsIdRotateSecretPost200Response>(_ctx))!;
    }
}
